lib/IRGen/GenObjC.cpp

```diff
@@ -1723,6 +1723,21 @@ void IRGenFunction::emitForeignReferenceTypeLifetimeOperation(
       cast<llvm::FunctionType>(llvmFn->getFunctionType())->getParamType(0);
   value = Builder.CreateBitCast(value, argType);
 
+  // Check if the pointer is null.
+  auto nullValue = llvm::Constant::getNullValue(argType);
+  auto hasValue = Builder.CreateICmpNE(value, nullValue);
+
+  auto nonNullValueBB = createBasicBlock("lifetime.nonnull-value");
+  auto contBB = createBasicBlock("lifetime.cont");
+
+  // If null, just continue.
+  Builder.CreateCondBr(hasValue, nonNullValueBB, contBB);
+
+  // If non-null, emit a call to release/retain function.
+  Builder.emitBlock(nonNullValueBB);
   auto call = Builder.CreateCall(llvmFn->getFunctionType(), llvmFn, value);
   call->setDoesNotThrow();
+  Builder.CreateBr(contBB);
```

Comment:
Do we not need to put anything into this block? Will that just be the default insertion point for the builder after you emit it so everything works out?

Reply:
Yeap, exactly, `Builder.emitBlock(contBB)` on the line below will move the insertion point to the newly inserted continuation block

Reply:
Never mind, this make sense to me.